src/rebar3_hex_publish.erl

```diff
@@ -174,38 +135,161 @@ format_error(Reason) ->
     rebar3_hex_error:format_error(Reason).
 
 %% ===================================================================
-%% Public API
+%% Private
 %% ===================================================================
 
-publish(App, HexConfig, State) ->
-    Name = rebar_app_info:name(App),
 
-    Version = rebar_app_info:original_vsn(App),
-    ResolvedVersion = rebar_utils:vcs_vsn(App, Version, State),
-    {application, _, AppDetails} = rebar3_hex_file:update_app_src(App, ResolvedVersion),
+%% ===================================================================
+%% Publish package only operations
+%% ===================================================================
+handle_task(#{args := #{task := package, app := undefined}, multi_app := true}) ->
+    ?RAISE({app_switch_required, "--app required when publishing with the package argument in a umbrella"});
+
+handle_task(#{args := #{task := package}, apps := [App]} = Task) ->
+    maybe_warn_about_single_app_args(Task),
+    #{args := Args, repo := Repo, state := State} = Task,
+    publish_package(State, Repo, App, Args);
+
+handle_task(#{args := #{task := package, app := AppName}, apps := Apps} = Task) ->
+    #{args := Args, repo := Repo, state := State} = Task,
+    case rebar3_hex_app:find(Apps, AppName) of
+        {error, app_not_found} ->
+            ?RAISE({app_not_found, AppName});
+        {ok, App} ->
+            publish_package(State, Repo, App, Args)
+    end,
+    {ok, State};
+
+%% ===================================================================
+%% Publish docs only operations
+%% ===================================================================
+%% TODO: Move doc publish code into this module
+handle_task(#{args := #{task := docs, app := undefined}, multi_app := true}) ->
+    ?RAISE({app_switch_required, "--app required when running publish docs only in a umbrella"});
 
+handle_task(#{args := #{task := docs}, apps := [App]} = Task) ->
+    #{args := Args, repo := Repo, state := State} = Task,
+    publish_docs(State, Repo, App, Args),
+    {ok, State};
 
-    Deps = rebar_state:get(State, {locks, default}, []),
-    {TopLevel, Excluded} = gather_deps(Deps),
+handle_task(#{args := #{task := docs, app := AppName}, apps := Apps} = Task) ->
+    case rebar3_hex_app:find(Apps, AppName) of
```

Comment:
Rationale : If you specified an app to publish or revert, let's go ahead and check if it actually exists.